code/game/machinery/washing_machine.dm

```diff
@@ -57,283 +118,307 @@
-			return
-		if( istype(W,/obj/item/clothing/suit/syndicatefake ) )
-			to_chat(user, "This item does not fit.")
-			return
-//		if( istype(W,/obj/item/clothing/suit/powered ) )
-//			to_chat(user, "This item does not fit.")
-//			return
-		if( istype(W,/obj/item/clothing/suit/cyborg_suit ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/suit/bomb_suit ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/suit/armor ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/mask/gas ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/mask/cigarette ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/head/syndicatefake ) )
-			to_chat(user, "This item does not fit.")
-			return
-//		if( istype(W,/obj/item/clothing/head/powered ) )
-//			to_chat(user, "This item does not fit.")
-//			return
-		if( istype(W,/obj/item/clothing/head/helmet ) )
-			to_chat(user, "This item does not fit.")
-			return
-		if( istype(W,/obj/item/clothing/gloves/furgloves ) )
-			to_chat(user, "This item does not fit.")
-			return
+	else
+		var/list/allowed = list(/obj/item/stack/sheet/hairlesshide, W,/obj/item/clothing/under, /obj/item/clothing/mask, /obj/item/clothing/head, /obj/item/clothing/gloves, /obj/item/clothing/shoes, /obj/item/clothing/suit, /obj/item/bedsheet, /obj/item/storage/wallet)
+		var/list/prohibited = list(/obj/item/clothing/suit/space, /obj/item/clothing/suit/syndicatefake, /obj/item/clothing/suit/bomb_suit, /obj/item/clothing/suit/armor, /obj/item/clothing/mask/gas, /obj/item/clothing/head/syndicatefake, /obj/item/clothing/head/helmet, /obj/item/clothing/suit/cyborg_suit, /obj/item/clothing/mask/cigarette)
+		var/goahead // there has to be a better way to do this
+		for(var/I in 1 to length(prohibited))
+			if(istype(W, allowed[I]))
+				goahead = TRUE
+		if(!goahead)
+			return ..()
+		if(!istype(W, (/obj/item/clothing/head/helmet/space/plasmaman)))
+			for(var/I in 1 to length(prohibited))
+				if(istype(W, prohibited[I]))
+					to_chat(user, "[W] is not washing machine safe.")
+					return
 		if(istype(W, /obj/item/clothing/gloves/color/black/krav_maga/sec))
 			to_chat(user, "<span class='warning'>Washing these gloves would fry the electronics!</span>")
 			return
-		if(W.flags & NODROP) //if "can't drop" item
-			to_chat(user, "<span class='notice'>\The [W] is stuck to your hand, you cannot put it in the washing machine!</span>")
+		if(W.flags && NODROP) //if "can't drop" item
+			to_chat(user, "<span class='notice'>\The [W] is stuck to your hand, you cannot put it in [src]!</span>")
 			return
 
-		if(contents.len < 5)
-			if( state in list(1, 3) )
-				user.drop_item()
-				W.loc = src
-				state = 3
+		if(length(contents) < 10)
+			var/mob/living/M
+			if(!M) // no placing items in the machine if theres a mob inside
```

Comment:
this if statement will always return TRUE, if you don't specificy the value of a var you initialize, it will initialize as null